pkg/tuned/tuned.go

```diff
@@ -896,32 +904,51 @@ func (c *Controller) changeWatcher() (err error) {
 	for {
 		select {
 		case <-c.stopCh:
-			// Termination signal received, stop
-			klog.V(2).Infof("changeWatcher done")
-			if err := c.tunedStop(nil); err != nil {
-				klog.Errorf("%s", err.Error())
-			}
+			klog.Infof("termination signal received, stop")
+
 			return nil
 
 		case s := <-sockConns:
+			const socketCmdStop = "stop"
+			var rolledBack bool
+
 			if s.err != nil {
 				return fmt.Errorf("connection accept error: %v", err)
 			}
 
-			buf := make([]byte, len("stop"))
+			buf := make([]byte, len(socketCmdStop))
 			nr, _ := s.conn.Read(buf)
 			data := buf[0:nr]
 
-			if string(data) == "stop" {
-				if err := c.tunedStop(&s); err != nil {
-					klog.Errorf("%s", err.Error())
-				}
-				return nil
+			if string(data) != socketCmdStop {
+				// We only support one command over the socket interface at this point
```

Comment:
maybe you can log `string(data)`, it's a case that should not happen, so if ever it occurs, better have it in the logs :)

Reply:
Fixed.